### PR TITLE
Assignment2

### PR DESCRIPTION
**Pull request by HulpeMiruna**

### CODE REVIEW

File: Assignment 2/Hulpe Dalia - Miruna/BookStore/BookStore/Controllers/AccountsController.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            await _userManager.AddToRoleAsync(user, "Administrator");
+
+            return RedirectToAction(nameof(Admin));
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Employee(string searchInput = null, string searchField = null)
+        {
+            var booksModel = new List<BookModel>();
+            switch (searchField)
```
Comment (Alex-Cosma) [inline Assignment 2/Hulpe Dalia - Miruna/BookStore/BookStore/Controllers/AccountsController.cs:93]: I don't know much about .NET but this surely isn't what I wanted
I want you to combine the search fields, not search on just one or the other.
My expectation is that you do something like `_bookLogic.search(searchInput)` which would basically check all those columns (title, author, genre) and retrieve all books which have either of them match the search input

Comment (Alex-Cosma) [reply Assignment 2/Hulpe Dalia - Miruna/BookStore/BookStore/Controllers/AccountsController.cs:93]: Also, why do I find `book` logic in `AccountsController`? Why is it not in ... `BooksController` ?

**Review by Alex-Cosma — COMMENTED** [review]
And you're missing quite a bit of tests

File: Assignment 2/Hulpe Dalia - Miruna/BookStore/BookStore/Models/BookViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace BookStore.Models
+{
+    public class BookViewModel
```
Comment (balazsmark888) [inline Assignment 2/Hulpe Dalia - Miruna/BookStore/BookStore/Models/BookViewModel.cs:6]: Validation mismatch:
The field restrictions should be present on the Database Entity as well (StringLength, Range, Required). 
For ex.: right now the Title, Author and Genre can be null in the database and can have any length.

Tip: You can use these same **Attributes** on the BookEntity to achieve this (new migration should be created afterwards).

File: Assignment 2/Hulpe Dalia - Miruna/BookStore/BookStore.BusinessLogic/BookStore.BusinessLogic.csproj [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Aspose.Cells" Version="22.4.0" />
+    <PackageReference Include="Aspose.PDF" Version="22.3.0" />
+    <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="3.2.0" />
+    <PackageReference Include="CsvHelper" Version="27.2.1" />
+    <PackageReference Include="GemBox.Document" Version="33.0.1483" />
+    <PackageReference Include="IronPdf" Version="2022.4.5455" />
+    <PackageReference Include="Magnum" Version="2.1.3" />
```
Comment (balazsmark888) [inline Assignment 2/Hulpe Dalia - Miruna/BookStore/BookStore.BusinessLogic/BookStore.BusinessLogic.csproj:14]: Aspose.Cells, Magnum, OpenHtmlToPdf and sautinsoft.pdffocus are deprecated and not used in the project; recommend removing the dependencies.

**Review by balazsmark888 — COMMENTED** [review]
There are a lot of unused **using statements** within the project, would recommend removing them.

**Pull request merged**